tests/test_spider.py

```diff
@@ -200,6 +200,31 @@ def test_load_pa11y_rules_file(tmpdir):
     assert result == expected_result
 
 
+def test_single_url():
```

Comment:
Could you check your coverage numbers? Have you covered the code you introduced? (I don't know...just asking. One test seems like it's too little but any scenario I can think of would likely be covered by other existing tests.)